schemas/v1.0/azure.yaml.json

```diff
@@ -135,9 +140,125 @@
                                 "description": ""
                             }
                         }
+                    },
+                    "scripts": {
+                        "type": "object",
+                        "title": "Definition of command hooks for developer integration",
+                        "descriptions": "Command hooks should match `azd` command names prefixed with `pre` or `post` depending on when the script should execute relative to the command",
+                        "minProperties": 1,
+                        "additionalProperties": {
+                            "type": "object",
+                            "additionalProperties": false,
+                            "properties": {
+                                "type": {
+                                    "type": "string",
+                                    "title": "Type of script",
+                                    "description": "Optional. The type of script to use for the command hook. (Default: bash)",
+                                    "enum": [
+                                        "bash",
+                                        "powershell"
+                                    ]
+                                },
+                                "location": {
+                                    "type": "string",
+                                    "title": "The location of the script",
+                                    "description": "Optional. The script can reference a file system path or be declared inline. (Default: path)",
+                                    "enum": [
+                                        "path",
+                                        "inline"
+                                    ]
+                                },
+                                "path": {
+                                    "type": "string",
+                                    "title": "Path to the command hook script relative to the azure.yaml file",
+                                    "description": "When `location` is set to `path` contains the full relative path to the command hook to execute"
+                                },
+                                "script": {
```

Comment:
Do we have a strong use case for inline scripts? Could we simplify things by only supporting external scripts for now?

Reply:
+1.
IMO, having `inline scripts` brings a lot of noisy lines to the `azure.yaml`, combining project configuration with the inline scripts

Reply:
You don't need to use them, but inline scripts for smaller workloads can be easier to manage vs external scripts.  This is only for extensibility and is in parity with something like package.json in node or any azdo/github pipelines.